vaccine_feed_ingest_schema/location.py

```diff
@@ -221,22 +255,46 @@ class OpenDate(BaseModel):
     }
     """
 
-    opens: Optional[str]
-    closes: Optional[str]
+    opens: Optional[StringDate]
+    closes: Optional[StringDate]
+
+    @root_validator
+    @classmethod
+    def validate_closes_after_opens(cls, values: dict) -> dict:
+        opens = values.get("opens")
+        closes = values.get("closes")
+
+        if opens and closes:
+            if closes < opens:
+                raise ValueError("Closes date must be after opens date")
+
+        return values
 
 
 class OpenHour(BaseModel):
     """
     {
         "day": str as day of week enum e.g. monday,
-        "opens": str as hh:mm,
-        "closes": str as hh:mm,
+        "opens": str as 24h hh:mm,
+        "closes": str as 24h hh:mm,
     }
     """
 
     day: DayOfWeek
-    opens: str
-    closes: str
+    opens: StringTime
+    closes: StringTime
```

Comment:
To confirm: we plan to compute the timezone based on the location on-display, so these should be in local time, right?

Reply:
Yes, these should be in local time.

Reply:
We can add a timezone field if we need to later